### ISSUE TITLE
First access failed because authentication

### ISSUE THREAD

**Issue by aungyin** [sdiaz/FOSRestBundleByExample#19]
Error Message is as below,

{"code":401,"message":"Unauthorized"}

What is the problem here. Can you suggest any solution? Please.

**Comment by toniIlic** [comment]
Before reading this you should know that I haven't been successful with this approach and had opened a ticket on this case here: https://github.com/sdiaz/FOSRestBundleByExample/issues/20.
You should authenticate to view the content so because this bundle is using mopa/wsse-authentication-bundle I think sending a X-WSSE header with login info should suffice. You can take a look at http://daniel-pomrehn.de/2014/05/21/en_symfony2-wsse-webservices/ under "Test the Api" for explanation on how to send a request with WSSE header.

**Comment by SakyaStelios** [comment]
I have the problem too. I go to the token section, generate the token and I get this;

```js
{
  "X-WSSE": "UsernameToken Username=\"admin\", PasswordDigest=\"1lqkTW0dLJkdfcz5BkQE7kM/VpY=\", Nonce=\"NGEyZDNmYzcwYTIxZDUxNg==\", Created=\"2015-05-21T00:12:13-05:00\""
}
```

Then, I go to /api/v1/users.json and set the headers:

```
X-wsse: UsernameToken Username="admin", PasswordDigest="Wxn1WmPFn2b7zkKwr3K7YKHvqNw=", Nonce="MDdmYTY5ZjFiM2UzYzg1ZA==", Created="2015-05-21T00:15:55-05:00"
Authorization: WSSE profile="UsernameToken"
Accept: application/json
```

Then I send a get request. But I have an 401 error. What is the correct flow?

**Comment by SakyaStelios** [comment]
I create a pull request to solve it (ad44506)

**Comment by kahllac** [comment]
thx for the fix but now i have 403 Forbidden with salt.json. Any idea ?

**Comment by sdiaz** [comment]
I have merged (ad44506)

**Comment by wujku** [comment]
31de04c3ef5a93e2f76eefcd84869f3e2d61c37b the problem occurs again.

```
{"code":401,"message":"You are not authenticated"}
```

**Comment by robbanl** [comment]
Ping @sdiaz

**Comment by robbanl** [comment]
@wujku Did you ever solve this?

**Comment by KhanMaytok** [comment]
It can be solved modifing the firewall route

**Comment by robbanl** [comment]
@SakyaStelios Tried modifying like in your previous pull request, but still same error.

**Comment by jokari4242** [comment]
I have same error. 
How can it be solved ? Thanks

**Comment by skyjooy** [comment]
I have the same error. Any help or solution ? Thx

**Comment by skyjooy** [comment]
@robbanl Do you have find a issue ? I have the same problem. Thx

**Comment by robbanl** [comment]
@skyjooy Nope, unfortunately not :( I remember I changed bundle to some other bundle, but can't remember what bundle :(
